### PR TITLE
GEODE-9758: Configure locator serialization filtering by default on Java 8

### PR DESCRIPTION
**Pull request by jchen21**
### For all changes:
- [ ] Is there a JIRA ticket associated with this PR? Is it referenced in the commit message?

- [ ] Has your PR been rebased against the latest commit within the target branch (typically `develop`)?

- [ ] Is your initial contribution a single, squashed commit?

- [ ] Does `gradlew build` run cleanly?

- [ ] Have you written or updated unit tests to verify your changes?

- [ ] If adding new dependencies to the code, are these dependencies licensed in a way that is compatible for inclusion under [ASF 2.0](http://www.apache.org/legal/resolved.html#category-a)?

### CODE REVIEW

File: geode-core/src/main/java/org/apache/geode/distributed/LocatorLauncher.java [hunk 1]
```diff
@@ -709,11 +722,31 @@ private boolean isStartable() {
    * @see org.apache.geode.distributed.AbstractLauncher.Status#ONLINE
    * @see org.apache.geode.distributed.AbstractLauncher.Status#STARTING
    */
-  @SuppressWarnings("deprecation")
   public LocatorState start() {
     if (isStartable()) {
       INSTANCE.compareAndSet(null, this);
 
+      boolean serializationFilterConfigured = false;
+      if (isJavaVersionAtLeast(JAVA_1_8) && isJavaVersionAtMost(JAVA_1_8)) {
```
Comment (dschneider-pivotal) [inline geode-core/src/main/java/org/apache/geode/distributed/LocatorLauncher.java:730]: shouldn't this use the new "isJavaVersion" method you added?

File: geode-core/src/main/java/org/apache/geode/distributed/LocatorLauncher.java [hunk 2]
```diff
@@ -729,6 +762,17 @@ public LocatorState start() {
               bindAddress, true, getDistributedSystemProperties(),
               getHostnameForClients(),
               Paths.get(workingDirectory));
+          if (isJavaVersionAtLeast(JAVA_1_8) && isJavaVersionAtMost(JAVA_1_8)
```
Comment (dschneider-pivotal) [inline geode-core/src/main/java/org/apache/geode/distributed/LocatorLauncher.java:765]: shouldn't this use the new "isJavaVersion" method you added?

**Review by dschneider-pivotal — DISMISSED** [review]

File: geode-core/src/main/java/org/apache/geode/internal/InternalDataSerializer.java [hunk 3]
```diff
@@ -447,6 +378,11 @@ static void clearSerializationFilter() {
     serializationFilter = defaultSerializationFilter;
   }
 
+  @VisibleForTesting
+  public static ObjectInputFilter getSerializationFilter() {
```
Comment (kirklund) [inline geode-core/src/main/java/org/apache/geode/internal/InternalDataSerializer.java:382]: Can this be non-public or does a test outside this package need to call it?

File: geode-core/src/main/java/org/apache/geode/internal/lang/SystemUtils.java [hunk 4]
```diff
@@ -50,6 +55,16 @@
 
   private static final String LINE_SEPARATOR = System.getProperty("line.separator");
 
+  /**
+   * Is the Java version the requested version.
+   *
+   * @param requiredVersion the required version
+   * @return true if the actual version is the required version
+   */
+  public static boolean isJavaVersion(final JavaVersion requiredVersion) {
```
Comment (kirklund) [inline geode-core/src/main/java/org/apache/geode/internal/lang/SystemUtils.java:64]: I'm leaning towards removing `isJavaVersion` and just have all callers use Apache Commons' JavaVersion directly.

File: geode-core/src/test/java/org/apache/geode/distributed/LocatorLauncherTest.java [hunk 5]
```diff
@@ -30,7 +32,7 @@
 public class LocatorLauncherTest {
 
   @Test
-  public void canBeMocked() throws Exception {
+  public void canBeMocked() {
```
Comment (kirklund) [inline geode-core/src/test/java/org/apache/geode/distributed/LocatorLauncherTest.java:35]: I think you should revert the changes to this file. Nothing here is pertinent to the PR.

File: geode-core/src/test/java/org/apache/geode/distributed/internal/ServerLocationTest.java [hunk 6]
```diff
@@ -0,0 +1,41 @@
...
+
+import static org.assertj.core.api.Assertions.assertThat;
+
+import org.junit.Test;
+import org.junit.experimental.categories.Category;
+
+import org.apache.geode.test.junit.categories.MembershipTest;
+
+@Category(MembershipTest.class)
+public class ServerLocationTest {
```
Comment (kirklund) [inline geode-core/src/test/java/org/apache/geode/distributed/internal/ServerLocationTest.java:25]: What is this test doing in this PR? I think it may have come from a merge. We should probably revert this and delete it from the PR.

File: geode-serialization/src/main/java/org/apache/geode/internal/serialization/filter/SerializableObjectConfig.java [hunk 7]
```diff
@@ -0,0 +1,28 @@
...
+
+  default String getFilterPatternIfEnabled() {
+    return getValidateSerializableObjects() ? getSerializableObjectFilter() : null;
+  }
+
+  boolean getValidateSerializableObjects(); // validate-serializable-objects -- boolean
+
+  void setValidateSerializableObjects(boolean value); // set validate-serializable-objects
+
+  String getSerializableObjectFilter(); // serializable-object-filter -- string
```
Comment (kirklund) [inline geode-serialization/src/main/java/org/apache/geode/internal/serialization/filter/SerializableObjectConfig.java:27]: Delete the comments from the end of the lines

File: geode-serialization/src/test/java/org/apache/geode/internal/serialization/filter/ReflectionObjectInputFilterApiTest.java [hunk 8]
```diff
@@ -0,0 +1,231 @@
...
+    } else if (isJavaVersionAtLeast(JAVA_1_8)) {
+      api = new ReflectionObjectInputFilterApi(ApiPackage.SUN_MISC);
+    }
+  }
+
+  // @Test
+  // public void constructsWithoutThrowing() {
+  // assertThatCode(() -> new
+  // ReflectionObjectInputFilterApi(apiPackage)).doesNotThrowAnyException();
+  // }
```
Comment (kirklund) [inline geode-serialization/src/test/java/org/apache/geode/internal/serialization/filter/ReflectionObjectInputFilterApiTest.java:55]: Delete

**Review by kirklund — CHANGES_REQUESTED** [review]
I tried to add comments to every class or test name that should change for consistency. Just try to make sure everything uses a consistent naming.

There is one class where we need a TODO comment added that we'll need to resolve on the next pass-through of reviewing everything.

**Review by DonalEvans — APPROVED** [review]
Approving the files for which I'm a code owner.

File: geode-core/src/main/java/org/apache/geode/distributed/LocatorLauncher.java (hunk 1 above)
Comment (jchen21) [reply geode-core/src/main/java/org/apache/geode/distributed/LocatorLauncher.java:730]: We have removed `isJavaVersion`. And keep `isJavaVersionAtLeast` and `isJavaVersionAtMost`, which are more convenient to handle multiple Java versions, instead of a specific version.

File: geode-core/src/main/java/org/apache/geode/distributed/LocatorLauncher.java (hunk 2 above)
Comment (jchen21) [reply geode-core/src/main/java/org/apache/geode/distributed/LocatorLauncher.java:765]: We have removed isJavaVersion. And keep isJavaVersionAtLeast and isJavaVersionAtMost, which are more convenient to handle multiple Java versions, instead of a specific version.

**Review by jinmeiliao — APPROVED** [review]

**Review by kirklund — APPROVED** [review]

**Review by onichols-pivotal — APPROVED** [review]

**Review by mhansonp — APPROVED** [review]

**Review by gesterzhou — APPROVED** [review]

**Review by Bill — APPROVED** [review]
Approved!

**Review by agingade — APPROVED** [review]
